diff -Naurp a/Lib/ensurepip/__init__.py b/Lib/ensurepip/__init__.py
--- a/Lib/ensurepip/__init__.py	2015-02-25 05:27:44.000000000 -0600
+++ b/Lib/ensurepip/__init__.py	2015-05-19 10:31:19.023778636 -0500
@@ -1,16 +1,20 @@
 import os
 import os.path
+import pip
 import pkgutil
+import setuptools
+import shutil
 import sys
 import tempfile
+from ensurepip import rewheel
 
 
 __all__ = ["version", "bootstrap"]
 
 
-_SETUPTOOLS_VERSION = "12.0.5"
+_SETUPTOOLS_VERSION = setuptools.version.__version__
 
-_PIP_VERSION = "6.0.8"
+_PIP_VERSION = pip.__version__
 
 # pip currently requires ssl support, so we try to provide a nicer
 # error message when that is missing (http://bugs.python.org/issue19744)
@@ -38,6 +42,8 @@ def _run_pip(args, additional_paths=None
 
     # Install the bundled software
     import pip
+    if args[0] in ["install", "list", "wheel"]:
+        args.append('--pre')
     pip.main(args)
 
 
@@ -87,20 +93,39 @@ def bootstrap(*, root=None, upgrade=Fals
         # omit pip and easy_install
         os.environ["ENSUREPIP_OPTIONS"] = "install"
 
+    whls = []
+    rewheel_dir = None
+    # try to see if we have system-wide versions of _PROJECTS
+    dep_records = rewheel.find_system_records([p[0] for p in _PROJECTS])
+    # TODO: check if system-wide versions are the newest ones
+    # if --upgrade is used?
+    if all(dep_records):
+        # if we have all _PROJECTS installed system-wide, we'll recreate
+        # wheels from them and install those
+        rewheel_dir = tempfile.TemporaryDirectory()
+        for dr in dep_records:
+            new_whl = rewheel.rewheel_from_record(dr, rewheel_dir.name)
+            whls.append(os.path.join(rewheel_dir.name, new_whl))
+    else:
+        # if we don't have all the _PROJECTS installed system-wide,
+        # let's just fall back to bundled wheels
+        for project, version in _PROJECTS:
+            whl = os.path.join(
+                os.path.dirname(__file__),
+                "_bundled",
+                "{}-{}-py2.py3-none-any.whl".format(project, version)
+            )
+            whls.append(whl)
+
     with tempfile.TemporaryDirectory() as tmpdir:
         # Put our bundled wheels into a temporary directory and construct the
         # additional paths that need added to sys.path
         additional_paths = []
-        for project, version in _PROJECTS:
-            wheel_name = "{}-{}-py2.py3-none-any.whl".format(project, version)
-            whl = pkgutil.get_data(
-                "ensurepip",
-                "_bundled/{}".format(wheel_name),
-            )
-            with open(os.path.join(tmpdir, wheel_name), "wb") as fp:
-                fp.write(whl)
-
-            additional_paths.append(os.path.join(tmpdir, wheel_name))
+        for whl in whls:
+            shutil.copy(whl, tmpdir)
+            additional_paths.append(os.path.join(tmpdir, os.path.basename(whl)))
+        if rewheel_dir:
+            rewheel_dir.cleanup()
 
         # Construct the arguments to be passed to the pip command
         args = ["install", "--no-index", "--find-links", tmpdir]
diff -Naurp a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2015-02-25 05:27:45.000000000 -0600
+++ b/Makefile.pre.in	2015-05-19 09:51:41.283355029 -0500
@@ -1159,7 +1159,7 @@ LIBSUBDIRS=	tkinter tkinter/test tkinter
 		test/test_asyncio \
 		collections concurrent concurrent/futures encodings \
 		email email/mime test/test_email test/test_email/data \
-		ensurepip ensurepip/_bundled \
+		ensurepip ensurepip/_bundled ensurepip/rewheel \
 		html json test/test_json http dbm xmlrpc \
 		sqlite3 sqlite3/test \
 		logging csv wsgiref urllib \
